src/riscv64/interrupt.c

```diff
@@ -245,6 +259,17 @@ void trap_interrupt(void)
     }
     int_debug("   calling runloop\n");
     runloop();
+  exit_fault:
+    console("cpu ");
+    print_u64(ci->id);
+    console(", state ");
+    console(state_strings[ci->state]);
+    console(", vector ");
+    print_u64(v);
+    console("\n");
+    dump_context(ctx);
+    // apic_ipi(TARGET_EXCLUSIVE_BROADCAST, ICR_ASSERT, shutdown_vector);
```

Comment:
Ok, I added another commit which implements multiple IPI vectors per CPU (using a per-CPU bitmask of pending IPI vectors) and changes these commented-out lines to calls to send_ipi(). This will avoid the "flush_handler" kludge and TLB shootdowns on unrelated (e.g. wakeup) IPIs.